In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

In [2]:
file1 = "../../Data_Collection/Census_API/sample.csv"
file2 = "../../Data_Collection/Census_API/CensusDataFrameClean.csv"

sample = pd.read_csv(file1)
tot_pop = pd.read_csv(file2)

In [3]:
sample.head()

Unnamed: 0                 County           State  Population  Median Age  \
0        1831       Strafford County   New Hampshire    125913.0        37.1   
1        1913         Madison County        New York     72089.0        41.1   
2        3185  Las Piedras Municipio     Puerto Rico     38466.0        37.6   
3        2297       Tillamook County          Oregon     25552.0        48.1   
4          32         Lincoln County        Arkansas     13935.0        37.6   

   Household Income  Per Capita Income  Poverty Count  Poverty Rate  \
0           63533.0            30598.0        11875.0      9.431115   
1           55858.0            26824.0         7932.0     11.003066   
2           20119.0             9906.0        18254.0     47.454895   
3           43777.0            23688.0         3821.0     14.953820   
4           32369.0            12610.0         2158.0     15.486186   

   Unemployed Citizens  
0               3953.0  
1               1700.0  
2               2715.0  
3                789.0  
4                431.0

In [4]:
tot_pop.head()

Unnamed: 0                Name  Population  Median Age  Household Income  \
0           0   Carroll, Arkansas     27690.0        44.4           38145.0   
1           1    Chicot, Arkansas     11189.0        43.7           29628.0   
2           2     Clark, Arkansas     22684.0        33.6           35595.0   
3           3      Clay, Arkansas     15202.0        44.0           32404.0   
4           4  Cleburne, Arkansas     25587.0        47.9           41717.0   

   Per Capita Income  Poverty Count  Poverty Rate  Unemployed Citizens  \
0            20888.0         4789.0     17.295052                739.0   
1            18853.0         3252.0     29.064260                446.0   
2            19500.0         4762.0     20.992770                858.0   
3            19192.0         3340.0     21.970793                474.0   
4            25078.0         3901.0     15.246023                903.0   

   Men in Law-Enforcement  Women in Law-Enforcement  Law-Enforcement per 1000  \
0                    91.0                       4.0                  3.430841   
1                   131.0                      17.0                 13.227277   
2                    94.0                      12.0                  4.672897   
3                    76.0                       0.0                  4.999342   
4                    98.0                       9.0                  4.181811   

   Unemployment Rate (%)  
0               2.668834  
1               3.986058  
2               3.782402  
3               3.118011  
4               3.529136

In [7]:
sample["County"] = sample["County"] + ", " + sample["State"]
sample["County"] = sample["County"].str.replace(" County","")
sample.head()

Unnamed: 0                               County           State  \
0        1831            Strafford,  New Hampshire   New Hampshire   
1        1913                   Madison,  New York        New York   
2        3185  Las Piedras Municipio,  Puerto Rico     Puerto Rico   
3        2297                   Tillamook,  Oregon          Oregon   
4          32                   Lincoln,  Arkansas        Arkansas   

   Population  Median Age  Household Income  Per Capita Income  Poverty Count  \
0    125913.0        37.1           63533.0            30598.0        11875.0   
1     72089.0        41.1           55858.0            26824.0         7932.0   
2     38466.0        37.6           20119.0             9906.0        18254.0   
3     25552.0        48.1           43777.0            23688.0         3821.0   
4     13935.0        37.6           32369.0            12610.0         2158.0   

   Poverty Rate  Unemployed Citizens  
0      9.431115               3953.0  
1     11.003066               1700.0  
2     47.454895               2715.0  
3     14.953820                789.0  
4     15.486186                431.0

In [10]:
sample.drop(columns="State")
sample.head()

Unnamed: 0                               County           State  \
0        1831            Strafford,  New Hampshire   New Hampshire   
1        1913                   Madison,  New York        New York   
2        3185  Las Piedras Municipio,  Puerto Rico     Puerto Rico   
3        2297                   Tillamook,  Oregon          Oregon   
4          32                   Lincoln,  Arkansas        Arkansas   

   Population  Median Age  Household Income  Per Capita Income  Poverty Count  \
0    125913.0        37.1           63533.0            30598.0        11875.0   
1     72089.0        41.1           55858.0            26824.0         7932.0   
2     38466.0        37.6           20119.0             9906.0        18254.0   
3     25552.0        48.1           43777.0            23688.0         3821.0   
4     13935.0        37.6           32369.0            12610.0         2158.0   

   Poverty Rate  Unemployed Citizens  
0      9.431115               3953.0  
1     11.003066               1700.0  
2     47.454895               2715.0  
3     14.953820                789.0  
4     15.486186                431.0

In [12]:
sample_stat = pd.DataFrame({"County":sample["County"],"Population":sample["Population"]})
tot_pop_stat = pd.DataFrame({"County":tot_pop["Name"],"Population":tot_pop["Population"]})


In [22]:
sample_stat.head()

County  Population  Expected Population
0            Strafford,  New Hampshire    125913.0             85754.54
1                   Madison,  New York     72089.0             85754.54
2  Las Piedras Municipio,  Puerto Rico     38466.0             85754.54
3                   Tillamook,  Oregon     25552.0             85754.54
4                   Lincoln,  Arkansas     13935.0             85754.54

In [23]:
tot_pop_stat.head()

County  Population  Expected Population
0   Carroll, Arkansas     27690.0        100027.188509
1    Chicot, Arkansas     11189.0        100027.188509
2     Clark, Arkansas     22684.0        100027.188509
3      Clay, Arkansas     15202.0        100027.188509
4  Cleburne, Arkansas     25587.0        100027.188509

In [18]:
sample_ev = sample_stat["Population"].mean()
tot_pop_ev = tot_pop_stat["Population"].mean()

In [19]:
sample_ev

85754.54

In [20]:
tot_pop_ev

100027.18850931677

In [21]:
tot_pop_stat["Expected Population"] = tot_pop_ev
sample_stat["Expected Population"] = sample_ev

In [24]:
crit_val = stats.chi2.ppf(q = 0.95, df = 2)
crit_val

5.991464547107979

In [29]:
stats.chisquare(sample_stat["Population"], sample_stat['Expected Population'])

Power_divergenceResult(statistic=441152198.4769208, pvalue=0.0)

In [30]:
stats.ttest_ind(sample_stat["Population"], tot_pop_stat["Population"], equal_var=False)

Ttest_indResult(statistic=-1.2178382765109728, pvalue=0.22357197864533)

We are 95% confident that our sample population is statistically similar to the entire population.

p-value = 0.223572
p-value > .05